### PR TITLE
Feat: re-write of terraform-applier

### PR DESCRIPTION
**Pull request by asiyani**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,162 @@
...
+docker-buildx: test ## Build and push docker image for the manager for cross-platform support
+	# copy existing Dockerfile and insert --platform=${BUILDPLATFORM} into Dockerfile.cross, and preserve the original Dockerfile
+	sed -e '1 s/\(^FROM\)/FROM --platform=\$$\{BUILDPLATFORM\}/; t' -e ' 1,// s//FROM --platform=\$$\{BUILDPLATFORM\}/' Dockerfile > Dockerfile.cross
+	- docker buildx create --name project-v3-builder
+	docker buildx use project-v3-builder
+	- docker buildx build --push --platform=$(PLATFORMS) --tag ${IMG} -f Dockerfile.cross .
+	- docker buildx rm project-v3-builder
+	rm Dockerfile.cross
+
+##@ Deployment
```
Comment (george-angel) [inline Makefile:99]: Where did this monster Makefile come from, and what does this "deploy" section do? When would I use it?

File: .gitignore [hunk 2]
```diff
@@ -1 +1,27 @@
...
+*.test
+
+# Output of the go coverage tool, specifically when used with LiteIDE
+*.out
+
+# Kubernetes Generated files - skip generated files, except for vendored files
+
+!vendor/**/zz_generated.*
+
+# editor and IDE paraphernalia
```
Comment (george-angel) [inline .gitignore:23]: These don't belong in repository ignore, but in developer's global ignore. The project does not care about every editor thats out there.

File: .gitignore [hunk 3]
```diff
@@ -1 +1,27 @@
 terraform-applier
+
+# Binaries for programs and plugins
+*.exe
+*.exe~
```
Comment (george-angel) [inline .gitignore:5]: We don't build exe or dll - why are these here?

File: hack/boilerplate.go.txt
Comment (george-angel) [inline hack/boilerplate.go.txt:1]: What is this, generated?

File: main.go [hunk 4]
```diff
@@ -146,117 +169,166 @@ func terraformVersionString(ctx context.Context, execPath string) (string, error
 	return version.String(), nil
 }
 
+func getKubeClient() (*kubernetes.Clientset, error) {
```
Comment (george-angel) [inline main.go:172]: https://go.dev/doc/effective_go#Getters

File: manifests/base/namespaced/leader_election_role_binding.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: terraform-applier-leader-election
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: terraform-applier-leader-election
+subjects:
+  - kind: ServiceAccount
+    name: terraform-applier
+    namespace: terraform-applier
```
Comment (george-angel) [inline manifests/base/namespaced/leader_election_role_binding.yaml:12]: I think its better to omit "namespace" here, it will default to "current", which is less surprising behaviour.

File: webserver/webserver.go [hunk 6]
```diff
@@ -18,13 +18,15 @@ var staticFiles embed.FS
 //go:embed templates/status.html
 var statusHTML string
 
+var log hclog.Logger
```
Comment (george-angel) [inline webserver/webserver.go:21]: Not a big deal, but instead of every package having to define it's own logger, you can create it once in local "log" package and import that everywhere else and use the single instance.

File: Makefile (hunk 1 above)
Comment (asiyani) [reply Makefile]: yes agree. this was work in progress i am now using go generate and reduced make file to normal basic steps

File: webserver/webserver.go (hunk 6 above)
Comment (asiyani) [reply webserver/webserver.go:21]: in this PR we have removed webserver for now. this line is here just to stop package erroring. in next PR we need to modify webserver package to work with new crd.

**Review by george-angel — APPROVED** [review]
